ladybug/windrose.py

```diff
@@ -0,0 +1,953 @@
+
+        Default is a legend parameter with minimum and maximum values of the
+        windrose analysis values. If show_stack is False, then these min and max
+        will be the average of the binned values per orientation.
+        The default segment count is 10.
+        """
+        if self._legend_parameters is None:
+            if self.show_stack:
+                min_analysis = self.analysis_data_collection.min
+                max_analysis = self.analysis_data_collection.max
+            else:
+                # If not stacked histogram, ensure legend parameter range is defined
+                # by wind mean range per orientation.
+                wind_means = [sum(bin)/float(len(bin)) for bin in self.data
+                              if len(bin) > 0]
+                min_analysis = min(wind_means)
+                max_analysis = max(wind_means)
+            segment_count = 10
+            self._legend_parameters = LegendParameters(min=min_analysis,
+                                                       max=max_analysis,
+                                                       segment_count=segment_count)
+        return self._legend_parameters
+
+    @legend_parameters.setter
+    def legend_parameters(self, legend_parameters):
+        """Set the legend parameters for the windrose plot.
+
+        Args:
+            legend_parameters: A LegendParameters object.
+        """
+        legtyp = type(LegendParameters(0, 1, 10))
+        assert isinstance(legend_parameters, legtyp), 'legend_parameters must be' \
+            ' a LegendParameters. Got {}.'.format(type(legend_parameters))
+        self._legend_parameters = legend_parameters
+
+    @property
+    def legend(self):
+        """The Legend object for this plot"""
+        if self._legend is None:
+            self._legend = self._container._legend
+        return self._legend
+
+    @legend.setter
+    def legend(self, legend):
+        """Set the Legend object"""
+        assert isinstance(legend, type(Legend())), 'legend must be a Legend. ' \
+            'Got {}.'.format(type(legend))
+        self._legend = legend
+
+    @property
+    def direction_data_collection(self):
+        """The direction data collection assigned to this windrose plot."""
+        return self._direction_data_collection
+
+    @property
+    def analysis_data_collection(self):
+        """The analysis data collection assigned to this windrose plot."""
+        return self._analysis_data_collection
+
+    @property
+    def analysis_values(self):
+        """ The hourly wind analysis values to bin by direction (i.e wind speed)."""
+        return self._analysis_data_collection.values
+
+    @property
+    def direction_values(self):
+        """The direction data values in this windrose plot."""
+        return self._direction_data_collection.values
+
+    @property
+    def data(self):
+        """A histogram of wind analysis values binned by the wind directions."""
+        return self._data
```

Comment:
This looks dangerous to me. Isn't the data a list? So a user could call this public property, append a bunch of stuff to this list and then the class has been mutated / messed up without any check. I would suggest either making this property hidden or using tuples instead of lists so that users can mutate the class.

Reply:
I'll make it into tuples. We need this exposed so people can access the histogram.